### PR TITLE
Focus not set when clicking on cell margin

### LINKED ISSUE #12437: Focus not set when clicking on cell margin
## Description

When clicking on a cell margin to select a cell without changing it to edit mode, the keyboard shortcuts don't seem to work for the cell. My guess is that the notebook document is not actually focused (i.e., what should happen in command mode) when clicking on a cell margin.

## Reproduce

1. Create an ipywidgets slider
2. Drag the widgets handle (thereby focusing the slider handle DOM element)
3. Click on the margin of a cell
4. Press "A" to create a cell above. This doesn't work.
5. Click inside the cell, thereby focusing its editor
6. Now click on the margin and the "A" keyboard shortcut works.

focus

## Expected behavior

Clicking on the cell margin in command mode focuses the notebook DOM element, which enables keyboard shortcuts for the cell.

## Context

- Operating System and version: macos 12.3.1
- Browser and version: Firefox 99
- JupyterLab version: 3.3.4

### PR DESCRIPTION
**Pull request by fcollonval**
## References

Fixes #12437

## Code changes

Focus the notebook node if the active cell changes but does not contain the focused element.

## User-facing changes

Fixes the issue

## Backwards-incompatible changes

N/A

### CODE REVIEW

**Comment by fcollonval** [comment]
@jasongrout would you mind reviewing this one?

File: packages/notebook/src/widget.ts [hunk 1]
```diff
@@ -1918,7 +1918,12 @@ export class Notebook extends StaticNotebook {
         activeCell.editor.focus();
       }
     }
-    if (force && !this.node.contains(document.activeElement)) {
+    if (
+      (force && !this.node.contains(document.activeElement)) ||
+      // Focus the notebook if the active cell changes but as not the focus
+      // Xref: https://github.com/jupyterlab/jupyterlab/issues/12437
```
Comment (afshin) [inline packages/notebook/src/widget.ts:1924]:
```suggestion
```

File: packages/notebook/src/widget.ts [hunk 2]
```diff
@@ -1918,7 +1918,12 @@ export class Notebook extends StaticNotebook {
         activeCell.editor.focus();
       }
     }
-    if (force && !this.node.contains(document.activeElement)) {
+    if (
+      (force && !this.node.contains(document.activeElement)) ||
+      // Focus the notebook if the active cell changes but as not the focus
```
Comment (afshin) [inline packages/notebook/src/widget.ts:1923]:
```suggestion
      // Focus notebook if active cell changes but does not have focus.
```

**Review by afshin — APPROVED** [review]
Fixes https://github.com/jupyterlab/jupyterlab/issues/12437

Thank you!

**Comment by afshin** [comment]
cc: @jasongrout

File: packages/notebook/src/widget.ts (hunk 1 above)
Comment (afshin) [reply packages/notebook/src/widget.ts:1924]: I think we can just leave the "Fixes ..." comment in this PR without linking to it in the comments, what do you think?

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.4.x

**Pull request merged**